File: AMF5/src/test/scala/scalaPlatform/RenderDomainElementTest.scala
package scalaPlatform

import amf.apicontract.client.scala.model.domain.{Payload, Response}
import amf.apicontract.client.scala.OASConfiguration
import amf.core.client.platform.model.DataTypes
import amf.shapes.client.scala.model.domain.ScalarShape
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should.Matchers
import org.yaml.model.YDocument
import org.yaml.render.JsonRender

import java.util.Arrays.asList

class RenderDomainElementTest extends AsyncFlatSpec with Matchers with FileReader {

  "AMFElementClient" should "emit a domain element" in {
    val payload = Payload()
      .withMediaType("application/json")
      .withSchema(
        ScalarShape()
          .withDataType(DataTypes.Boolean)
          .withId("aScalar"))
      .withId("somethingElse")
    val response = Response()
      .withStatusCode("401")
      .withDisplayName("My Example Response")
      .withDescription("An example response")
      .withPayloads(List(payload))
      .withId("someId")
    val client = OASConfiguration.OAS30().elementClient()
    val emittedNode = client.renderElement(response)
    val writtenString = JsonRender.render(emittedNode)
    val expected = readResource("/expected/emitted-response.json")
    writtenString shouldEqual expected
  }
}

File: AMF5/src/test/java/javaPlatform/RenderDomainElementTest.java
package javaPlatform;

import amf.apicontract.client.platform.AMFElementClient;
import amf.apicontract.client.platform.OASConfiguration;
import amf.apicontract.client.platform.model.domain.Payload;
import amf.apicontract.client.platform.model.domain.Response;
import amf.core.client.platform.model.DataTypes;
import amf.shapes.client.platform.model.domain.ScalarShape;
import org.junit.Test;
import org.yaml.builder.JsonOutputBuilder;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.Writer;
import java.nio.file.Files;
import java.util.Date;

import static java.util.Collections.*;
import static javaPlatform.StringEquals.assertIgnoringWhitespace;

public class RenderDomainElementTest implements FileReader{

    @Test
    public void renderDomainElement() throws IOException {
        File tmpFile = File.createTempFile("rendered-response.json", String.valueOf(new Date().getTime()));
        Writer writer = new FileWriter(tmpFile);
        JsonOutputBuilder<Writer> jsonBuilder = JsonOutputBuilder.apply(writer, false);
        AMFElementClient client = OASConfiguration.OAS30().elementClient();
        Payload payload = new Payload()
                .withMediaType("application/json")
                .withSchema(new ScalarShape().withDataType(DataTypes.Boolean()).withId("aScalar"))
                .withId("somethingElse");
        Response response = (Response) new Response()
                .withStatusCode("401")
                .withDisplayName("My Example Response")
                .withDescription("An example response")
                .withPayloads(singletonList(payload))
                .withId("someId");
        client.renderToBuilder(response, jsonBuilder);
        writer.flush();
        writer.close();
        String goldenContent = readResource("/expected/rendered-response.json");
        String writtenContent = String.join("", Files.readAllLines(tmpFile.toPath()));
        assertIgnoringWhitespace(writtenContent, goldenContent);
    }
}
